code/modules/antagonists/changeling/powers/chameleon_skin.dm

```diff
@@ -1,25 +1,43 @@
 /datum/action/changeling/chameleon_skin
-	name = "Chameleon Skin"
-	desc = "Our skin pigmentation rapidly changes to suit our current environment. Costs 25 chemicals."
-	helptext = "Allows us to become invisible after a few seconds of standing still. Can be toggled on and off."
+	name = "Darkness Adaptation"
+	desc = "Our skin pigmentation and eyes rapidly changes to suit the darkness. Needs 25 chemicals in-hand to toggle."
+	helptext = "Allows us to darken and change the translucency of our pigmentation, and adapt our eyes to see in dark conditions, \
+	The translucent effect works best in dark enviroments. Can be toggled on and off."
 	button_icon_state = "chameleon_skin"
 	dna_cost = 2
 	chemical_cost = 25
 	req_human = TRUE
+	/////is ability active?
+	var/is_active = FALSE
 
 /datum/action/changeling/chameleon_skin/sting_action(mob/user)
-	var/mob/living/carbon/human/H = user //SHOULD always be human, because req_human = TRUE
-	if(!istype(H)) // req_human could be done in can_sting stuff.
-		return
+	var/mob/living/carbon/human/cling = user //SHOULD always be human, because req_human = TRUE
 	..()
-	if(H.dna.get_mutation(/datum/mutation/human/chameleon))
-		H.dna.remove_mutation(/datum/mutation/human/chameleon)
+	if(is_active == FALSE)
+		animate(cling, alpha = 85,time = 3 SECONDS)
+		cling.visible_message("<span class='warning'>[cling.name] skin is suddenly slowly becoming translucent!</span>", \
+						"<span class='notice'>You are now far more stealthy and better at looking in the dark.</span>")
+		animate(cling, color = COLOR_GRAY, time = 3 SECONDS) //darkens their overall appearance
+		var/obj/item/organ/internal/eyes/eyes = cling.get_organ_by_type(/obj/item/organ/internal/eyes)
+		eyes.lighting_cutoff = LIGHTING_CUTOFF_MEDIUM // Adds barely usable, kinda shit night vision
+		eyes.flash_protect = max(eyes.flash_protect - 1, FLASH_PROTECTION_HYPER_SENSITIVE) //Reduces flash protection by one level.
+		cling.update_sight()
+		is_active = TRUE
 	else
-		H.dna.add_mutation(/datum/mutation/human/chameleon)
-	return TRUE
+		disable_ability(cling)
+		is_active = FALSE
 
 /datum/action/changeling/chameleon_skin/Remove(mob/user)
-	if(user.has_dna())
-		var/mob/living/carbon/C = user
-		C.dna.remove_mutation(/datum/mutation/human/chameleon)
 	..()
+	var/mob/living/carbon/human/cling = user
+	disable_ability(cling)
+
+/datum/action/changeling/chameleon_skin/proc/disable_ability(mob/living/carbon/human/cling) //Restores adaptation
+	animate(cling, alpha = 255, time = 3 SECONDS)
+	cling.visible_message("<span class='warning'>[cling.name] appears from thin air!</span>", \
+					"<span class='notice'>You are now appearing normal and lost the ability at looking in the dark.</span>")
```

Comment:
Few things: 
- Span macros
- Changings use "We" and not "You" 
- You don't need `.name`
- Minor grammar check 
```suggestion
	cling.visible_message(
		span_warning("[cling] appears from thin air!)",
		span_notice("We are now appearing normal and lost the ability to see in the dark."),
	)
```